From 94143f8b06ca3ef41ad0a70c6c6d0d96672975b5 Mon Sep 17 00:00:00 2001
From: Ghislain MARY <ghislain.mary@belledonne-communications.com>
Date: Fri, 5 Dec 2014 10:31:19 +0100
Subject: [PATCH] Do not include devhelp and examples subdirs when not
 rebuilding docs.

---
 doc/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index dd39f69..29a69fc 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
+if REBUILD_DOCS
 SUBDIRS=devhelp examples
+endif
 
 # The top-level SGML file.
 DOC_MAIN_XML_FILE=gnome-xml.xml
-- 
2.1.3

