diff -urN libvpx-1.4.0.orig/build/make/gen_msvs_sln.sh libvpx-1.4.0/build/make/gen_msvs_sln.sh
--- libvpx-1.4.0.orig/build/make/gen_msvs_sln.sh	Fri Apr  3 20:49:19 2015
+++ libvpx-1.4.0/build/make/gen_msvs_sln.sh	Wed Aug 19 15:16:50 2015
@@ -227,7 +227,7 @@
 
   else
 $nows_sln_config: $outfile
-${TAB}\$(MSBUILD_TOOL) $outfile -m -t:Build \\
+${TAB}\$(MSBUILD_TOOL) $outfile -t:Build \\
 ${TAB}${TAB}-p:Configuration="$config" -p:Platform="$platform"
 
   endif
